### PR TITLE
Doc updated engine options and edit CLI page

### LINKED ISSUE #1039: Document updated engine-rpc options
Document new `engine-rpc` options that unify rpc ports:

- add `--engine-rpc-enabled` (required to use Besu as an execution client)
- add `--engine-rpc-port`
- remove `--engine-rpc-http-port`
- remove `--engine-rpc-ws-port`

see https://github.com/hyperledger/besu/pull/3834

### PR DESCRIPTION
**Pull request by alexandratran**
## Pull request checklist

Use the following list to make sure your PR fits the Besu documentation quality standard.

### Before creating the pull request

Make sure that:

- [x] [All commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] You've read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Documentation).
- [x] You've [previewed your changes locally](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation).

## Describe the change

Document updated engine options and make edits throughout CLI page for style consistency across boolean and host/port options.

## Issue fixed

fixes #1039 

## Impacted parts

For content changes:

- [x] Documentation content
- [ ] Documentation page organization

For tool changes:

- [ ] CircleCI workflow
- [ ] Build and QA tools (for example, lint or vale)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] Read the Docs configuration
- [ ] GitHub integration

## After creating your PR and tests have finished

Make sure that:

- [ ] You've fixed any issues raised by the tests.
- [ ] You've [previewed your changes on Read the Docs](https://wiki.hyperledger.org/display/BESU/Preview+the+documentation)
  and added a [preview link](#preview).

### CODE REVIEW

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -547,61 +565,67 @@ Contents of file must be at least 32 hex-encoded bytes and not begin with `0x`.
...
     ```bash
-    BESU_ENGINE_RPC_HTTP_PORT=8550
+    BESU_ENGINE_RPC_ENABLED=true
     ```
 
 === "Configuration file"
 
     ```bash
-    engine-rpc-http-port=8550
+    engine-rpc-port=true
```
Comment (jflo) [inline docs/Reference/CLI/CLI-Syntax.md:591]: should read engine-rpc-enabled

**Review by rolandtyler — COMMENTED** [review]
We should also update this page https://besu.hyperledger.org/en/stable/HowTo/Interact/APIs/Engine-API/

**Review by rolandtyler — APPROVED** [review]
lgtm

File: docs/HowTo/Interact/APIs/Engine-API.md [hunk 2]
```diff
@@ -9,20 +9,19 @@ These [API methods](../../../Reference/Engine-API-Methods.md) are a separate sub
...
-Also [enable the JSON-RPC](API.md#enabling-api-access).
-Ensure the [`ETH` method is enabled](Using-JSON-RPC-API.md#api-methods-enabled-by-default) (it's enabled by default).
+To configure the Engine API:
 
-### Service ports
-
-To specify the port the Engine API service listens on, use the
-[`--engine-rpc-http-port`](../../../Reference/CLI/CLI-Syntax.md#engine-rpc-http-port) or
-[`--engine-rpc-ws-port`](../../../Reference/CLI/CLI-Syntax.md#engine-rpc-ws-port) options.
+- Enable the Engine API using the [`--engine-rpc-enabled`](../../../Reference/CLI/CLI-Syntax.md#engine-rpc-enabled) option.
```
Comment (garyschulte) [inline docs/HowTo/Interact/APIs/Engine-API.md:14]: This is actually deprecated in the latest besu releases, and we instead defer to the presence or absence of `terminalTotalDifficulty` in the genesis config.  Relevant changes:

[engine-rpc-enabled parameter deprecated](https://github.com/hyperledger/besu/blob/main/besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java#L569-L574)

[log warning if engine-rpc-enabled used](https://github.com/hyperledger/besu/blob/main/besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java#L2115-L2119)

[merge config and therefore engine api deferring to genesis](https://github.com/hyperledger/besu/blob/main/besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java#L3192-L3201)

File: docs/Reference/CLI/CLI-Syntax.md [hunk 3]
```diff
@@ -547,61 +565,68 @@ Contents of file must be at least 32 hex-encoded bytes and not begin with `0x`.
...
-The listening port for the Engine API calls (`ENGINE`, `ETH`) for JSON-RPC over HTTP.
+Enables or disables the Engine JSON-RPC service.
+The default is `false`.
+
+!!! important
 
-### `engine-rpc-ws-port`
+    You must enable the Engine API to use Besu as an [execution client](../../Concepts/Merge.md#execution-clients) on
+    Ethereum Mainnet.
+
```
Comment (garyschulte) [inline docs/Reference/CLI/CLI-Syntax.md:601]: we should remove/rewrite this to indicate we are deferring to TTD for enabling engine-rpc-api

**Review by garyschulte — CHANGES_REQUESTED** [review]
recent changes invalidate some of these changes

File: docs/HowTo/Interact/APIs/Engine-API.md [hunk 4]
```diff
@@ -9,20 +9,18 @@ These [API methods](../../../Reference/Engine-API-Methods.md) are a separate sub
...
+- Specify the [host allowlist](#host-allowlist).
+- [Enable the JSON-RPC API](API.md#enable-api-access).
+  Ensure the [`ETH` method is enabled](Using-JSON-RPC-API.md#api-methods-enabled-by-default) (it's enabled by default).
 
-The default ports are:
+### Service ports
 
-* 8550 over HTTP.
-* 8551 over WebSocket.
+To specify the port the Engine API service listens on over HTTP and WebSocket, use the
```
Comment (garyschulte) [inline docs/HowTo/Interact/APIs/Engine-API.md:21]: suggest `...service listens on for HTTP and WebSockets ...`.  Nitpicky though

File: docs/HowTo/Interact/APIs/Engine-API.md [hunk 5]
```diff
@@ -9,20 +9,18 @@ These [API methods](../../../Reference/Engine-API-Methods.md) are a separate sub
...
+To configure the Engine API:
 
-### Service ports
-
-To specify the port the Engine API service listens on, use the
-[`--engine-rpc-http-port`](../../../Reference/CLI/CLI-Syntax.md#engine-rpc-http-port) or
-[`--engine-rpc-ws-port`](../../../Reference/CLI/CLI-Syntax.md#engine-rpc-ws-port) options.
+- Specify the [service ports](#service-ports).
+- Specify the [host allowlist](#host-allowlist).
+- [Enable the JSON-RPC API](API.md#enable-api-access).
```
Comment (garyschulte) [inline docs/HowTo/Interact/APIs/Engine-API.md:16]: it is possible I am just missing it, but where is this anchor?

**Review by garyschulte — COMMENTED** [review]
LGTM, just a nitpick and a question

File: docs/HowTo/Interact/APIs/Engine-API.md (hunk 5 above)
Comment (garyschulte) [reply docs/HowTo/Interact/APIs/Engine-API.md:14]: NVM, found it :)

**Review by garyschulte — APPROVED** [review]
🚢

**Review by jflo — APPROVED** [review]

**Pull request merged**